src/inc/til/some.h

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <array>
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    template<class T, size_t N>
+    class some
+    {
+    private:
+        std::array<T, N> _array;
+        size_t _used{ 0 };
+
+#ifdef UNIT_TESTING
+        friend class SomeTests;
+#endif
+
+    public:
+        using value_type = T;
+        using size_type = size_t;
+        using difference_type = ptrdiff_t;
+        using pointer = T*;
+        using const_pointer = const T*;
+        using reference = T&;
+        using const_reference = const T&;
+
+        using iterator = typename decltype(_array)::iterator;
+        using const_iterator = typename decltype(_array)::const_iterator;
+
+        using reverse_iterator = typename decltype(_array)::reverse_iterator;
+        using const_reverse_iterator = typename decltype(_array)::const_reverse_iterator;
+
+        void fill(const T& _Value)
+        {
+            _array.fill(_Value);
+            _used = N;
+        }
+
+        void swap(some& _Other) noexcept(std::is_nothrow_swappable<T>::value)
+        {
+            _array.swap(_Other._array);
+            std::swap(_used, _Other._used);
+        }
+
+        constexpr iterator begin() noexcept
+        {
+            return iterator(_array.data(), 0);
+        }
+
+        constexpr const_iterator begin() const noexcept
+        {
+            return const_iterator(_array.data(), 0);
+        }
+
+        constexpr iterator end() noexcept
+        {
+            return iterator(_array.data(), _used);

```

Comment:
and can this be return `_array.begin() + _used`? Or is that not cool

Reply:
seems ok.